arrays_lab.js

```diff
@@ -0,0 +1,235 @@
+    console.log(loopArr2[i])
+    i++
+}
+
+console.log('----')
+// 13. Write code that logs every value in an array using a for/of loop
+//
+// ```js
+// let logMeOutForOf = ["I", "am", "a", "happy", "array"]
+//
+// /* I
+//   am
+//   a
+//   happy
+//   array
+// */
+// ```
+let predArnold = ['get','to','the','chopper']
+for(let y of predArnold){
+    console.log(y)
+}
+
+
+// 14. Given an array of numbers, write code that logs the sum of all the numbers
+//
+// ```js
+// let firstArrToSum = [1, 2, 3, 4, 5, 6]
+// //  21
+//
+// let secondArrToSum = [1, 2, 3, 4, 5, -6]
+// //  9
+// ```
+let total=0
+let sumArray =[1, 2, 3, 4, 5, 6];
+for (let i = 0; i < sumArray.length; i++){
+       total+= sumArray[i]
+}
+console.log(total)
+
+
+// 15. Given an array of numbers, write code that logs all the odd numbers
+//
+// ```js
+//
+// let evenAndOddArr = [1,5,2,4,11,12,99,100]
+//
+// // 1, 5, 11, 99
+// ```
+let oddArr = [1,6,7,9,3,23,79,90,57,43,91]
+
+for (let i =0; i < oddArr.length; i++){
+      if (oddArr[i] % 2){
+        console.log(oddArr[i]);
+      }
+}
+console.log('----')
+// 16. Given an array of numbers, write code that logs the sum of all the even numbers
+//
+// ```js
+//
+// let evenAndOddArrToSum = [1,5,2,3,11,4,6]
+//
+// // 12
+// ```
+let sum=0;
+let evenAndOddArrToSum = [1,5,2,3,11,4,6,9,,60,59,46,37,23,14,90]
+for(let i =0; i< evenAndOddArrToSum.length; i++){
+      if (evenAndOddArrToSum[i] % 2 === 0){
+            sum += evenAndOddArrToSum[i]
+      }
+}console.log(sum)
+
+// 17. Write code such that noDupeList has all the same numbers as dupeFriendlyList, but has no more than one of each number.
+//
+// ```js
+let dupeFriendlyList = [4,2,6,2,2,6,4,9,2,1]
+ let noDupeList = []
+ let exac =[]
+for (let i =0; i < dupeFriendlyList.length; i++){
+
+    if (exac[i] === dupeFriendlyList[i]){
+        continue
+    } else {
+      console.log(noDupeList.push(exac[i]))
+    }
+}
+
+
+//noDupleList = [4, 2, 6, 9, 1]
+//
+// ```
+
+
+// 18. Make an array that contains all elements that appear exactly once in someRepeatsAgain
+//
+// ```js
+ let someRepeatsAgain = [25,11,30,31,50,28,4,37,13,20,24,38,28,14,44,33,7,43,39,35,36,42,1,40,7,14,23,46,21,39,11,42,12,38,41,48,20,23,29,24,50,41,38,23,11,30,50,13,13,16,10,8,3,43,10,20,28,39,24,36,21,13,40,25,37,39,31,4,46,20,38,2,7,11,11,41,45,9,49,31,38,23,41,16,49,29,14,6,6,11,5,39,13,17,43,1,1,15,25]
+//
+// // [ 44,33,35,42,12,48,30,50,8,3,10,28,24,36,21,1000,40,37,4,46,20,2,7,45,9,31,38,23,41,16,49,29,14,6,11,5,39,13,17,43,1,15,25 ]
+let norep
+for (let i = 0; i <someRepeatsAgain.length; i++){
+
+}
+
+// ```
+//
+// 19. Given an array of strings, find the the string with the most "a"s in it.
+//
+// ```js
+// let mostAStrings = ["apes", "abba", "apple"]
+//
+// //"abba"
+// ```
+
+
+
+// 20. Find the second smallest number in an Array of Ints
+//
+// ```js
+let secondSmallestArr = [11, 52, 10, 7, 50, 46, 79, 78, 13, 26, 83, 92, 89, 81, 1, 41, 4, 23, 57, 41, 80, 83, 41, 69]
```

Comment:
This is one way to solve this, but what are other ways that you can solve this not using sort function? Interviewers will ask questions like this for white boarding process.